lib/scope.js

```diff
@@ -679,7 +679,22 @@ module.exports = {
     for ( let data_i = 0; data_i < fields_to_set.length; data_i++ ) {
       let { handle, tag, type, row } = fields_to_set[ data_i ];
       if ( tag === 'BUTTON' && type === 'submit' ) {
+        let page_url = await scope.page.url();
         await scope.setVariable( scope, { handle, set_to: row.value });
+      
+        // Have to catch errors right now or may get stuck in an infinite loop?
+        // TODO: Don't catch errors - maybe the developer expected/wanted an error
```

Comment:
I think this will be a useful future step. 

In Python testing, the tests always throw errors. If they are expected, the user can say something like: 
```python
def test_passes():
    with pytest.raises(Exception) as e_info:
        x = 1 / 0
```
If the user doesn't realize an error will happen, then the test will fail. 
```python
def test_fails():
    x = 1 / 0
``` 
We could try something analogous. 

BUT that seems like a nice-to-have feature, not a must-have feature right now (at least to me).

Reply:
Good thoughts and issue made!